packages/push/src/push_vm/stack.rs

```diff
@@ -362,7 +359,11 @@ impl<T> Stack<T> {
     {
         let iter = iter.into_iter();
         // Check that adding these items won't overflow the stack.
-        if iter.len() + self.size() > self.max_stack_size {
+        if iter
+            .len()
+            .checked_add(self.size())
+            .is_some_and(|x| x > self.max_stack_size)
```

Comment:
This doesn't seem quite right to me. If `iter.len().checked_add(self.size())` returns `None`, then that means that adding the current stack size and the new items will be larger than `usize`, and it seems that that should fail in some way. Am I missing something?

Reply:
good catch! this should probably be .is_none_or instead